troposphere/emr.py

```diff
@@ -0,0 +1,126 @@
+# Copyright (c) 2012-2013, Antonio Alonso Dominguez <alonso.domin@gmail.com>
+# All rights reserved.
+#
+# See LICENSE file for full license.
+
+from . import AWSObject, AWSProperty, AWSHelperFn
+from .validators import (boolean, integer)
+
+
+class KeyValue(AWSHelperFn):
+    def __init__(self, key, value):
+        self.data = {'Key': key, 'Value': value}
+
+    def JSONrepr(self):
+        return self.data
+
+
+class Application(AWSProperty):
+    props = {
+        'AdditionalInfo': (dict, False),
+        'Args': (list, False),
+        'Name': (basestring, False),
+        'Version': (basestring, False)
+    }
+
+
+class BootstrapActionConfig(AWSProperty):
+    props = {
+        'Name': (basestring, True),
+        'ScriptBootstrapAction': (basestring, True)
+    }
+
+
+class ClusterConfiguration(AWSProperty):
+    props = {
+        'Classification': (basestring, False),
+        'ConfigurationProperties': (dict, False),
+        'Configurations': (list, False)
+    }
+
+
+class InstanceGroupConfigProperty(AWSProperty):
+    props = {
+        'BidPrice': (basestring, False),
+        'Configurations': (list, False),
```

Comment:
Mind pointing to an example of how to express a list in which each item must be custom validated? Added the `emr_configuration_property` for the `ConfigurationProperties` as it seems to be a list of _colon-separated key-value pairs_ but if I express it as:

```python
'ConfigurationProperties': ([emr_configuration_property], False)
```

I get the following error when running the tests:

```
TypeError: isinstance() arg 2 must be a class, type, or tuple of classes and types
```

It works fine if it weren't a list - it validates the value using the regex.